Engine.UnitTests/MixinTests.cs

```diff
@@ -260,7 +261,50 @@ public void TestParameterizeAndRemoveMixin()
             var paramMember3 = TypeData.GetTypeData(plan1).GetMember("B");
             // now this should be null.
             Assert.IsNull(paramMember3);
+        }
+
+        public class ActionStep : TestStep
+        {
+            public Action action { get; set; }
+            
+            public override void Run()
+            {
+                action();
+            }
+            public static ActionStep FromAction(Action action) => new ActionStep()
+            {
+                action = action
+            };
+        }
+        
+        [Test]
+        public void CannotAddMixinWhilePlanRunning()
+        {
+            ManualResetEvent continueEvt = new ManualResetEvent(false);
+            ManualResetEvent started = new ManualResetEvent(false);
+            var step = ActionStep.FromAction(() =>
+            {
+                started.Set();
+                continueEvt.WaitOne();
+            });
+            var plan = new TestPlan();
+            plan.ChildTestSteps.Add(step);
+            var planRun = plan.ExecuteAsync();
+            started.WaitOne();
+            // ok, now the test plan is waiting until we set continueEvt.
 
+            var a = AnnotationCollection.Annotate(step);
+            var addMixin = a.GetIcon(IconNames.AddMixin);
+            Assert.IsNotNull(addMixin);
+            var enabled = addMixin.GetAll<IEnabledAnnotation>();
+            Assert.IsTrue(enabled.Any());
+            var isDisabled = enabled.Any(x => x.IsEnabled == false);
```

Comment:
```suggestion
            var isDisabled = enabled.All(x => x.IsEnabled == false);
```

Reply:
Actually it is disabled if any one of these are false, so the Any is more correct.